ESP8266Interface.cpp

```diff
@@ -472,6 +478,33 @@ void ESP8266Interface::socket_attach(void *handle, void (*callback)(void *), voi
     _cbs[socket->id].data = data;
 }
 
+nsapi_error_t ESP8266Interface::setsockopt(nsapi_socket_t handle, int level,
+        int optname, const void *optval, unsigned optlen)
+{
+    struct esp8266_socket *socket = (struct esp8266_socket *)handle;
+
+    if(socket->connected == true) {// ESP8266 limitation, keepalive needs to be given before connecting
```

Comment:
Logically this test belongs inside the KEEPALIVE case - it won't apply generally.

Reply:
ok